mobile/src/board/hooks.tsx

```diff
@@ -0,0 +1,30 @@
+import { useContext, useEffect, useState } from "react";
+
+import { Board } from "./entity";
+import { BoardRepository } from "./repository";
+
+export function useBoard(initialState: Board): Board {
+  const [board, setBoard] = useState<Board>(initialState);
+
+  useEffect(() => {
+    const mercureURL = `/.well-known/mercure?topic=${encodeURI(
+      window.location.href
```

Comment:
I'd say that's fragile... This board can only be used on certain urls, and nothing in its API shows it. 

You'd better pass the topic to the hook, and leave the responsibility of reading it from the window location to the parent. 

In general: move side effects up the chain.